[WIP] Find out the commits that failed CorePatch

JingMatrix opened the pull request:
CorePatch only fails on certain devices (#32 ).
In this branch, I first revert the commit f3beb8688fc4d9e28d6a7123cae1a27bd039df17, which is known to cause the CorePatch issue on some devices.

After tests results from users for fore-coming commits, I will gradually figure out which commit in the master branch breaks CorePatch.

JingMatrix committed:
Update dependencies

DexBuilder is updated in the LSPlant repo.

Note: we should clean caches of GitHub Actions

JingMatrix committed:
Avoid navigating to app list before loaded

Close #27

JingMatrix committed:
Allow to de-activate log watchlog (#20)

Requested by #14, a switch is added to the release version.
Patch of msvamp rejected in original repository
mywalk added ui interface

Co-authored-by: mywalk <66966897+mywalkb@users.noreply.github.com>

JingMatrix committed:
Add commons-lang as a subproject

Avoid using a local version of this apache module

JingMatrix committed:
Add missing LD_LIBRARY_PATH env (#34)

Tests passed on Android 13 and Android 15 devices

JingMatrix commented:
@happypotter321 Now I have to use the most stupid way to find out which commit breaks CorePatch.
Please test the latest CI build and tell me if it works.

happypotter321 commented:
Just tried the latest Cl build (debug version) and followed the same steps as before, still didn't work for me unfortunately. Are the logs still needed btw? I saved it just in case, here it is. 
[LSPosed_2024-09-20T22_46_53.295508.zip](https://github.com/user-attachments/files/17078595/LSPosed_2024-09-20T22_46_53.295508.zip)

JingMatrix commented:
@happypotter321 It is okay if it still fails. The goal here is try to figure out which commit breaks CorePatch (that is my last resort).
Please be patient, if remains about four tests to reach the goal.
Attaching your logs is always easier for me, thanks.
Please try again the latest CI build and tell me your result.

happypotter321 commented:
Understood, same result with the build after the previous one I used. Here's the log, 
[LSPosed_2024-09-20T23_33_44.993648.zip](https://github.com/user-attachments/files/17079090/LSPosed_2024-09-20T23_33_44.993648.zip)

JingMatrix commented:
The GitHub Actions is quite buggy, please try my local build.
[LSPosed-4666b130.zip](https://github.com/user-attachments/files/17079729/LSPosed-4666b130.zip)

Edited: It turned out that I misunderstood how Github Actions works with pull-request. My fault...

happypotter321 commented:
That worked: :D 

[LSPosed_2024-09-21T00_49_36.606984.zip](https://github.com/user-attachments/files/17080022/LSPosed_2024-09-21T00_49_36.606984.zip)

JingMatrix commented:
So far, I found that CorePatch stops working for you once the following logs present:
```
[W/Magisk]: Failed to dlopen zygisk module: dlopen failed: library "libicu.so" not found: needed by /system/lib64/libharfbuzz_ng.so in namespace (default)
```
It seems that some dependency updates breaks CorePatch.

`libharfbuzz_ng.so` is simply a library for fonts, I don't see why LSPosed needed it.
I guess that it is commit 44570c5e that causes the above error

Please continue trying my local builds and upload your log.
[LSPosed-44570c5e.zip](https://github.com/user-attachments/files/17080767/LSPosed-44570c5e.zip)

happypotter321 commented:
Sure, and yes this one worked too.
[LSPosed_2024-09-21T02_48_43.732097.zip](https://github.com/user-attachments/files/17080917/LSPosed_2024-09-21T02_48_43.732097.zip)

JingMatrix commented:
That is cool, so my previous comment was wrong. It is not related to the `dlopen` failure of magisk.
Then I conjecture it was caused by the upstream update of LSPlant after commit d4e00eac

Please try this local build:
[LSPosed-d4e00eac.zip](https://github.com/user-attachments/files/17080981/LSPosed-d4e00eac.zip)

happypotter321 commented:
This one didn't work, yep. 
[LSPosed_2024-09-21T03_27_12.505228.zip](https://github.com/user-attachments/files/17081120/LSPosed_2024-09-21T03_27_12.505228.zip)

JingMatrix commented:
Then the bug might be caused by LSPlt, which confuses me a lot.
Please continue trying my local build, I wanna see if 75e30053 breaks CorePatch. Hoping that it will work.
[LSPosed-75e30053.zip](https://github.com/user-attachments/files/17083049/LSPosed-75e30053.zip)

happypotter321 commented:
This one didn't work, just tried it. 
[LSPosed_2024-09-21T11_44_53.637134.zip](https://github.com/user-attachments/files/17083103/LSPosed_2024-09-21T11_44_53.637134.zip)

pluseach commented:
same here, doesn't work too.. fail.

before installing an app: com.waze
[LSPosed_2024-09-21T13_21_29.864141-modules.zip](https://github.com/user-attachments/files/17083116/LSPosed_2024-09-21T13_21_29.864141-modules.zip)
[LSPosed_2024-09-21T13_21_19.406175-verbose.zip](https://github.com/user-attachments/files/17083118/LSPosed_2024-09-21T13_21_19.406175-verbose.zip)

after installing an app: com.waze
[LSPosed_2024-09-21T13_22_35.631262-modules.zip](https://github.com/user-attachments/files/17083122/LSPosed_2024-09-21T13_22_35.631262-modules.zip)
[LSPosed_2024-09-21T13_22_48.666308-verbose.zip](https://github.com/user-attachments/files/17083123/LSPosed_2024-09-21T13_22_48.666308-verbose.zip)

JingMatrix commented:
Okay. Then the bug is surely introduced by LSPlt, which might conflict ArtMethod de-optimizing.
To comfirm my conjecture, please also try the following build. It should work.
If so, the commits before 16c09cef are good.
[LSPosed-16c09cef.zip](https://github.com/user-attachments/files/17083134/LSPosed-16c09cef.zip)

pluseach commented:
yep it works, the app installed successfully.

happypotter321 commented:
Same here, app installed with that one. 
[LSPosed_2024-09-21T12_26_01.446840.zip](https://github.com/user-attachments/files/17083202/LSPosed_2024-09-21T12_26_01.446840.zip)

JingMatrix commented:
To temporarily fix this issue, I disable LSPlt in the latest commit.
Please try the latest build (debug version) and upload your logs.
I suppose that it should work without problems.

pluseach commented:
Thank you so much, CorePatch works normally now 🫡
1. WIP [10970539268](https://github.com/JingMatrix/LSPosed/actions/runs/10970539268)

before installing:
[LSPosed_2024-09-21T14_17_16.625255.zip](https://github.com/user-attachments/files/17083257/LSPosed_2024-09-21T14_17_16.625255.zip)
after installing:
[LSPosed_2024-09-21T14_18_36.698557.zip](https://github.com/user-attachments/files/17083260/LSPosed_2024-09-21T14_18_36.698557.zip)

JingMatrix commented:
@happypotter321 I need your log for the latest CI build debug version to better understand the problem.
If it works, you can keep using that version.
@pluseach Thanks for participating this issue. It would be better if you could use the **debug version** from latest CI build so that I could compare logs,

happypotter321 commented:
Yep, just tested it and it works.  And okay thanks.
[LSPosed_2024-09-21T13_47_00.007991.zip](https://github.com/user-attachments/files/17083407/LSPosed_2024-09-21T13_47_00.007991.zip)

pluseach commented:
@JingMatrix Of course, here's the logs:
Installing package com.waze
1. Latest Release Logs Version [10933177132](https://github.com/JingMatrix/LSPosed/actions/runs/10933177132): Installation Failed
[LSPosed_2024-09-21T15_26_56.888485-before_installing.zip](https://github.com/user-attachments/files/17083446/LSPosed_2024-09-21T15_26_56.888485-before_installing.zip)
[LSPosed_2024-09-21T15_28_13.820470-after_installing(fail).zip](https://github.com/user-attachments/files/17083448/LSPosed_2024-09-21T15_28_13.820470-after_installing.fail.zip)

Debug Logs Version:
[LSPosed_2024-09-21T16_07_01.317515_before-install.zip](https://github.com/user-attachments/files/17083515/LSPosed_2024-09-21T16_07_01.317515_before-install.zip)
[LSPosed_2024-09-21T16_09_09.174960_after-install(fail).zip](https://github.com/user-attachments/files/17083517/LSPosed_2024-09-21T16_09_09.174960_after-install.fail.zip)


2. Latest WIP Release Logs Version [10970714532](https://github.com/JingMatrix/LSPosed/actions/runs/10970714532): Installed Successfully
[LSPosed_2024-09-21T15_32_31.164139-before_installing.zip](https://github.com/user-attachments/files/17083455/LSPosed_2024-09-21T15_32_31.164139-before_installing.zip)
[LSPosed_2024-09-21T15_34_49.026593-after_installing(success).zip](https://github.com/user-attachments/files/17083456/LSPosed_2024-09-21T15_34_49.026593-after_installing.success.zip)

Debug Logs Version:
[LSPosed_2024-09-21T16_14_06.848650_before-install.zip](https://github.com/user-attachments/files/17083518/LSPosed_2024-09-21T16_14_06.848650_before-install.zip)
[LSPosed_2024-09-21T16_16_18.330281_after-install(success).zip](https://github.com/user-attachments/files/17083521/LSPosed_2024-09-21T16_16_18.330281_after-install.success.zip)


**_Sorry, I mistakenly mentioned a pull request._**

privacyguy123 commented:
This build has same number as builds that don't work - is it safe to download Actions build to test?

JingMatrix committed:
Fix Dobby hook fallback

Even if dladdr could not find function name for an address given by
art_symbol_resolver, we should still process inline hook.

JingMatrix commented:
@privacyguy123 Sure. The build name of LSPosed zip archive works in another way, only commits from the main branch will increase that number. We are in a pull-request, so the number stays the same.

JingMatrix commented:
Hello, everyone. In the latest build, I am try to actually fix the issue instead of a workaround. Please test it and upload the logs if it fails.

privacyguy123 commented:
> Hello, everyone. In the latest build, I am try to actually fix the issue instead of a workaround. Please test it and upload the logs if it fails.

Installing apps with differing signatures all good on my side now :)

privacyguy123 commented:
Never mind - you have introduced some regression with this build. Watchdog logs toggled off is now doing nothing and forcing log on/setting persists props every reboot again. I don't understand why this setting would be affected at all by new commits, I see the commit for it in the history. @JingMatrix

JingMatrix commented:
@privacyguy123 Thanks for reporting this issue.
It was my fault, I rebase the history with a change, and my device unfortunately didn't show that it breaks some stuff (on others' device). Already fixed in the latest build of master branch now.
